keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,912 @@
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As an end-user of Kubernetes, I'd like all my workload controllers to have consistent states
+
+#### Story 2
+As an developer building Kubernetes Operators, I'd like all my operators deployed to have
+consistent states.
+
+
+### Overview of all conditions
+
+The following table gives an overview on what conditions each of the workload resources support.
+
+|                                    | Progressing | Available |  ReplicaFailure | Suspended | Failed | Complete |
+| ---------------------------------  | ----------- | --------- | --------------- | --------- | ------ | -------- |
+| ReplicaSet & ReplicationController | -           | -         |  failed to create / delete pod (FailedCreate, FailedDelete)  | -         | -           | -        |
+| Deployment                         | True when scaling replicas / creating-updating new ReplicaSet / successfully finished progressing (Pods ready or available for MinReadySeconds). False when failed creating ReplicaSet / reached progressDeadlineSeconds. Unknown when rollout paused | True if if required number of replicas is available (takes MaxSurge and MaxUnavailable into account) | failure propagated from new or old ReplicaSet | -         | -*        | -*          |
+| StatefulSet                        | -           | -         | -               | -         | -      | -        |
+| DaemonSet                          | -           | -         | -               | -         | -      | -        |
+| Job                                | -           | -         | -               | True / False when suspended / resumed | failed execution  (BackoffLimitExceeded, DeadlineExceeded)| completed execution |
+| CronJob**                          | -           | -         | -               | -         | -      | -        |
+
+**\* Success of the rollout is instead represented by a Progressing condition (status and reason)**
+
+**\*\* CronJob does not even have Conditions field in its Status**
+
+### Proposed Conditions
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+We are proposing 4 new conditions to be added to the following controllers:
+- Available (DaemonSet, ReplicaSet & ReplicationController, StatefulSet)
+- Progressing (DaemonSet, StatefulSet)
+- Waiting (Job)
+- Running (Job)
```

Comment:
RS/RC can also be `Progressing` when creating pods, we have a batching mechanism, so it's possible that this will take some time, in which case reporting progressing seems reasonable.

Reply:
I am not sure about this. 
- so far we have profilled the Progressing condition as a way to present a rollout and this would break the familiarity.
- I think this is very similar to Available behaviour/cond which we are planning to intoduce

Reply:
Good point.